base/readpassphrase/readpassphrase.c

```diff
@@ -27,6 +27,10 @@
 #define _PATH_TTY "/dev/tty"
 #endif
 
+#if defined(__clang__) && __clang_major__ >= 13
```

Comment:
this fixes an error:
```
[3/2180] Building C object base/readpassphrase/CMakeFiles/readpassphrase.dir/readpassphrase.c.o
		FAILED: base/readpassphrase/CMakeFiles/readpassphrase.dir/readpassphrase.c.o 
		/usr/bin/clang-13 -DSTD_EXCEPTION_HAS_STACK_TRACE=1 -I../contrib/sentry-native/include -Iincludes/configs -I../base/readpassphrase/. -I../base/glibc-compatibility/memcpy -isystem ../contrib/libcxx/include -isystem ../contrib/libcxxabi/include -isystem ../contrib/libunwind/include -isystem ../contrib/libc-headers/x86_64-linux-gnu -isystem ../contrib/libc-headers -fdiagnostics-color=always  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -falign-functions=32  -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -Werror -Wframe-larger-than=65536 -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wno-c11-extensions -Wcomma -Wconditional-uninitialized -Wcovered-switch-default -Wdeprecated -Wembedded-directive -Wempty-init-stmt -Wextra-semi-stmt -Wextra-semi -Wgnu-case-range -Winconsistent-missing-destructor-override -Wnewline-eof -Wold-style-cast -Wrange-loop-analysis -Wredundant-parens -Wreserved-id-macro -Wshadow-field -Wshadow-uncaptured-local -Wshadow -Wstring-plus-int -Wundef -Wunreachable-code-return -Wunreachable-code -Wunused-exception-parameter -Wunused-macros -Wunused-member-function -Wzero-as-null-pointer-constant -Weverything -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-ctad-maybe-unsupported -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -O2 -g -DNDEBUG -O3  -fno-pie   -D OS_LINUX -Wno-unused-result -Wno-reserved-id-macro -Wno-disabled-macro-expansion -std=gnu11 -MD -MT base/readpassphrase/CMakeFiles/readpassphrase.dir/readpassphrase.c.o -MF base/readpassphrase/CMakeFiles/readpassphrase.dir/readpassphrase.c.o.d -o base/readpassphrase/CMakeFiles/readpassphrase.dir/readpassphrase.c.o   -c ../base/readpassphrase/readpassphrase.c
		/home/semin-serg/ClickHouse/base/readpassphrase/readpassphrase.c:138:32: error: identifier '__res' is reserved because it starts with '__' [-Werror,-Wreserved-identifier]
		                    ch = (char)tolower((unsigned char)ch);
		                               ^
		/home/semin-serg/ClickHouse/contrib/libc-headers/ctype.h:220:22: note: expanded from macro 'tolower'
		#  define tolower(c)    __tobody (c, tolower, *__ctype_tolower_loc (), (c))
		                        ^
		/home/semin-serg/ClickHouse/contrib/libc-headers/ctype.h:157:11: note: expanded from macro '__tobody'
		   ({ int __res;                                                              \
		          ^
		/home/semin-serg/ClickHouse/base/readpassphrase/readpassphrase.c:138:32: error: identifier '__c' is reserved because it starts with '__' [-Werror,-Wreserved-identifier]
		/home/semin-serg/ClickHouse/contrib/libc-headers/ctype.h:220:22: note: expanded from macro 'tolower'
		#  define tolower(c)    __tobody (c, tolower, *__ctype_tolower_loc (), (c))
		                        ^
		/home/semin-serg/ClickHouse/contrib/libc-headers/ctype.h:162:12: note: expanded from macro '__tobody'
		              int __c = (c);                                                  \
		                  ^
		/home/semin-serg/ClickHouse/base/readpassphrase/readpassphrase.c:140:32: error: identifier '__res' is reserved because it starts with '__' [-Werror,-Wreserved-identifier]
		                    ch = (char)toupper((unsigned char)ch);
		                               ^
		/home/semin-serg/ClickHouse/contrib/libc-headers/ctype.h:221:22: note: expanded from macro 'toupper'
		#  define toupper(c)    __tobody (c, toupper, *__ctype_toupper_loc (), (c))
		                        ^
		/home/semin-serg/ClickHouse/contrib/libc-headers/ctype.h:157:11: note: expanded from macro '__tobody'
		   ({ int __res;                                                              \
		          ^
		/home/semin-serg/ClickHouse/base/readpassphrase/readpassphrase.c:140:32: error: identifier '__c' is reserved because it starts with '__' [-Werror,-Wreserved-identifier]
		/home/semin-serg/ClickHouse/contrib/libc-headers/ctype.h:221:22: note: expanded from macro 'toupper'
		#  define toupper(c)    __tobody (c, toupper, *__ctype_toupper_loc (), (c))
		                        ^
		/home/semin-serg/ClickHouse/contrib/libc-headers/ctype.h:162:12: note: expanded from macro '__tobody'
		              int __c = (c);                                                  \
		                  ^
		4 errors generated.
```